src/pages/Payment.vue

```diff
@@ -77,15 +89,23 @@ export default {
         }
     },
     methods: {
+        showConfirmation: function() {
+            this.setMessage("Thanks, your payment request has been sent.", constants.MESSAGE_SUCCESS);
+            this.$router.message = this.message;
+            this.$router.push('/account');
+        },
         onRequestPayment() {
             Store.pendRecVouchers(
                 // on Success, route to /account
                 function(response) {
-                    this.$router.push('/account');
+                    this.showConfirmation();
                 }.bind(this),
                 // on Failure... hook for an alert?
                 function(error) {
-                }
+                    if (!Store.netMgr.online) {
```

Comment:
That sounds like a good plan 😄 I think we do it on Tap and Scan because we need to make sure that we're offline before we show queue specific messages.